src/impl/mpispace/Kokkos_MPISpace_AllocationRecord.cpp

```diff
@@ -60,7 +60,7 @@ SharedAllocationRecord<Kokkos::Experimental::MPISpace, void>::
     // Pass through allocated [ SharedAllocationHeader , user_memory ]
     // Pass through deallocation function
     : SharedAllocationRecord<void, void>(
-#ifdef KOKKOS_DEBUG
+#ifdef KOKKOS_ENABLE_DEBUG
```

Comment:
Would it make sense to use `KOKKOS_REMOTE_ENABLE_DEBUG` instead

Reply:
Also, we'd require to set KOKKOS_REMOTE_ENABLE_DEBUG in case of KOKKOS_ENABLE_DEBUG as some function signatures change in Kokkos.

Reply:
Why would you need to set `KOKKOS_ENABLE_DEBUG`? You may have a problem problem in the distributed case but everything is fine on the node and since most of the time is spent on-node you may want to keep that part in release mode.